### PR TITLE
Feature/doctor add working shifts

### PR DESCRIPTION
**Pull request by mohamedebrahim2020**

### CODE REVIEW

File: app/Http/Controllers/WorkingDayController.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    protected $doctorService;
+
+    public function __construct(DoctorService $doctorService)
+    {
+        $this->doctorService = $doctorService;
+    }
+    public function store(StoringDoctorWorkingDayRequest $request)
+    {
+        $workingDay = $this->doctorService->addWorkingDay($request->except('is_all_day'));
```
Comment (MohannadElemary2) [inline app/Http/Controllers/WorkingDayController.php]: Allow to send `is_all_day` for a specific day

File: app/Http/Requests/StoringDoctorWorkingDayRequest.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'day' => ['required', new EnumValue(WeekDays::class, false)],
+            'from' => 'nullable|required_with:to|date_format:g:a',
+            'to' => 'nullable|required_with:from|date_format:g:a|after:from',
```
Comment (MohannadElemary2) [inline app/Http/Requests/StoringDoctorWorkingDayRequest.php]: The doctor will send his working days on one request. Allow him to send an array of objects contains (day, from, to, all_day), validate them and store them in the database. It's not efficient to make user add his working shifts a shift by shift.

File: app/Http/Requests/StoringDoctorWorkingDayRequest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'day' => ['required', new EnumValue(WeekDays::class, false)],
+            'from' => 'nullable|required_with:to|date_format:g:a',
```
Comment (MohannadElemary2) [inline app/Http/Requests/StoringDoctorWorkingDayRequest.php]: The right dat format is `h:i A`

File: app/Http/Controllers/WorkingDayController.php (hunk 1 above)
Comment (mohamedebrahim2020) [reply app/Http/Controllers/WorkingDayController.php]: done

File: app/Http/Requests/StoringDoctorWorkingDayRequest.php (hunk 2 above)
Comment (mohamedebrahim2020) [reply app/Http/Requests/StoringDoctorWorkingDayRequest.php]: done

File: app/Http/Requests/StoringDoctorWorkingDayRequest.php (hunk 3 above)
Comment (mohamedebrahim2020) [reply app/Http/Requests/StoringDoctorWorkingDayRequest.php]: done

**Pull request merged**